pkg/reconciler/v1alpha1/autoscaling/kpa/kpa.go

```diff
@@ -285,6 +290,56 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	return nil
 }
 
+// HasRoute returns false if there is a route to a different revision but not this revision,
+// and true otherwise (including when unknown).
+func (c *Reconciler) HasRoute(pa *pav1alpha1.PodAutoscaler) bool {
+	routes, err := c.routes(pa)
+	if err != nil || len(routes) == 0 {
+		// Assume a route exists when running into error finding it. Even
+		// when no route is found, it may be set up.
+		return true
+	}
+	// alt becomes true if there is a route (traffic) to an alternative revision
+	// of the same service (or configuration).
+	alt := false
+	for _, route := range routes {
+		for _, t := range route.Status.Traffic {
```

Comment:
Should we check that it actually receives any traffic, i.e. percentage > 0? 

Because of the inverse logic, this loop is hard to read and comprehend, even though it does simple thing. Can you rewrite it? so you return the `alt`, rather than its inverse.  
Perhaps it would make it more readable if you have a per-route function.

Reply:
I didn't have a good name, but while explaining the algorithm to others, "replaced" came across as a better name than "hasRoute".  What do you think?